internal/cli/get.go

```diff
@@ -80,152 +80,101 @@ func NewCmdGet() *cobra.Command {
 			if cmd.Flags().Lookup("continue").Changed {
 				cont = &o.Continue
 			}
-			return RunGet(kind, name, labelSelector, fleetName, o.Output, limit, cont)
+			return RunGet(kind, name, labelSelector, owner, o.Output, limit, cont)
 		},
 		SilenceUsage: true,
 	}
 
-	cmd.Flags().StringVarP(&o.FleetName, "fleetname", "f", o.FleetName, "fleet name selector for listing devices")
+	cmd.Flags().StringVar(&o.Owner, "owner", o.Owner, "filter by owner")
 	cmd.Flags().StringVarP(&o.LabelSelector, "labelselector", "l", o.LabelSelector, "label selector as a comma-separated list of key=value")
 	cmd.Flags().StringVarP(&o.Output, "output", "o", o.Output, "output format (yaml)")
 	cmd.Flags().Int32Var(&o.Limit, "limit", o.Limit, "the maximum number of results returned in the list response")
 	cmd.Flags().StringVar(&o.Continue, "continue", o.Continue, "query more results starting from the value of the 'continue' field in the previous response")
 	return cmd
 }
 
-func RunGet(kind, name string, labelSelector, fleetName *string, output string, limit *int32, cont *string) error {
+func RunGet(kind, name string, labelSelector, owner *string, output string, limit *int32, cont *string) error {
 	c, err := client.NewFromConfigFile(defaultClientConfigFile)
 	if err != nil {
 		return fmt.Errorf("creating client: %v", err)
 	}
 
-	switch kind {
-	case "device":
-		if len(name) > 0 {
-			response, err := c.ReadDeviceWithResponse(context.Background(), name)
-			if err != nil {
-				return fmt.Errorf("reading device/%s: %v", name, err)
-			}
-			out, err := serializeResponse(response, err, fmt.Sprintf("device/%s", name))
-			if err != nil {
-				return fmt.Errorf("serializing response for device/%s: %v", name, err)
-			}
-			fmt.Printf("%s\n", string(out))
-		} else {
-			params := api.ListDevicesParams{
-				FleetName:     fleetName,
-				LabelSelector: labelSelector,
-				Limit:         limit,
-				Continue:      cont,
-			}
-			response, err := c.ListDevicesWithResponse(context.Background(), &params)
-			if err != nil {
-				return fmt.Errorf("listing devices: %v", err)
-			}
-			return printListResourceResponse(response, err, "devices", output)
+	var response interface{}
+
+	switch true {
```

Comment:
I love using a switch here very clean true is assumed in your && evaluations

```suggestion
	switch {
```